### PR TITLE
MD-5043 Disable s3 storage plugin if Minio is down. Move start_minio …

### PR DESCRIPTION
**Pull request by denysord88**
…script outside of the gen scripts block.

### CODE REVIEW

**Comment by denysord88** [comment]
@Agirish could you please review this PR? There are changes, that are fixing Functional tests failures when s3 tests are excluded or data generation is disabled.

File: framework/src/main/java/org/apache/drill/test/framework/TestDriver.java [hunk 1]
```diff
@@ -180,6 +180,18 @@ public int runTests() throws Exception {
   	  queryMemoryUsage();
     }
 
+    // Run Apache Minio server if needed for the S3 storage
+    if(cmdParam.excludeDependencies == null || !cmdParam.excludeDependencies.contains("s3")) {
+      Utils.startMinio();
+    } else {
+      // Disable S3 storage plugin if Minio server is down
```
Comment (Agirish) [inline framework/src/main/java/org/apache/drill/test/framework/TestDriver.java]: I think we should rename `s3` plugin as `s3test` or `minio` or something similar. If anyone has configured the S3 plugin to point to Amazon's S3, then we will disable or modify the plugin unintentionally.

Comment (denysord88) [reply framework/src/main/java/org/apache/drill/test/framework/TestDriver.java]: Done. Renamed "s3" to "s3minio".

File: framework/src/main/java/org/apache/drill/test/framework/TestDriver.java [hunk 2]
```diff
@@ -180,6 +180,18 @@ public int runTests() throws Exception {
...
 
+    // Run Apache Minio server if needed for the S3 storage
+    if(cmdParam.excludeDependencies == null || !cmdParam.excludeDependencies.contains("s3minio")) {
+      Utils.startMinio();
+    } else {
+      // Disable S3 storage plugin if Minio server is down
+      String templatePath = DrillTestDefaults.CWD + "/conf/plugin-templates/common/s3minio-storage-plugin.template";
+      boolean isSuccess = Utils.disableStoragePlugin(templatePath, "s3minio");
+      if(!isSuccess){
+        LOG.info("> Fail to disable S3 storage plugin");
```
Comment (Agirish) [inline framework/src/main/java/org/apache/drill/test/framework/TestDriver.java]: Update comments here and elsewhere to refer to the plugin by it's new name?

Comment (denysord88) [reply framework/src/main/java/org/apache/drill/test/framework/TestDriver.java]: Done.

**Comment by denysord88** [comment]
Successfully tested these changes on test-cluster.

**Pull request merged**